adyenv6b2ccheckoutaddon/acceleratoraddon/web/webroot/_ui/responsive/common/js/adyen.checkout.js

```diff
@@ -343,6 +345,54 @@ var AdyenCheckoutHybris = (function () {
             }
         },
 
+        initiateApple: function (amount, countryCode, applePayMerchantIdentifier, applePayMerchantName, label) {
+            var applePayNode = document.getElementById('adyen-component-button-container-' + label);
+            var adyenComponent = this.checkout.create("applepay", {
+                amount: {
+                    currency: amount.currency,
+                    value: amount.value
+                },
+                countryCode: countryCode,
+                configuration: {
+                    merchantName: applePayMerchantName,
+                    merchantIdentifier: applePayMerchantIdentifier
+                },
+                // Button config
+                buttonType: "plain",
+                buttonColor: "black",
+                onChange: (state, component) => {
+                    if (!state.isValid) {
+                        this.enablePlaceOrder(label);
+                    }
+                },
+                onSubmit: (state, component) => {
+                    if (!state.isValid) {
+                        this.enablePlaceOrder(label);
+                        return false;
+                    }
+                    this.makePayment(state.data.paymentMethod, component, this.handleResult, label);
+                },
+                onClick: (resolve, reject) => {
+                    if (this.isTermsAccepted(label)) {
+                        resolve();
+                    } else {
+                        reject();
+                        this.handleResult(ErrorMessages.TermsNotAccepted, true);
+                    }
+                }
+            });
+
+            adyenComponent.isAvailable()
+                .then(() => {
+                    adyenComponent.mount(applePayNode);
+                })
+                .catch(e => {
+                    // Apple Pay is not available
+                    console.log('Something went wrong trying to mount the Apple Pay component: ' + e);
+                    this.handleResult(ErrorMessages.PaymentNotAvailable, true);
```

Comment:
is it possible somehow to hide the entire payment method if Apple Pay is not available? (for Chrome/other browsers users)

Reply:
I think we can check browser type and enable it only for safari.

Reply:
done